docsource/installation.rst

```diff
@@ -0,0 +1,207 @@
+
+.. code-block:: bash
+
+    $ gedit [PATH_TO_YOUR_WORKSPACE_DIR]/deploy.cfg
+
+Make the following changes -
+
+.. code-block:: cfg
+
+    listeners = Search
+    listener-Search-class = us.kbase.workspace.modules.SearchPrototypeEventHandlerFactory
+    listener-Search-config-mongohost = localhost
+    listener-Search-config-mongodatabase = Search_test
+    listener-Search-config-mongouser = ""
+    listener-Search-config-mongopwd = ""
+
+4. Restart the Workspace Service. (See section on `Deploying the Workspace Service locally <https://github.com/kbase/workspace_deluxe/blob/dev-candidate/docsource/developers.rst>`_)
+
+5. Open a new terminal and save the following document as Empty.spec. Then load into ipython and save the object to the Workspace. Saving a new object will cause the Workspace listener to write a new event to the mongo instance.
+
+.. code-block:: javascript
+
+    module Empty {
+
+        /* @optional foo */
+        typedef structure {
+            int foo;
+        } AType;
+    };
+
+.. code-block:: bash
+
+    $ ipython
+
+    In [1]: spec = open("[PATH_TO_SPEC]/Empty.spec").read()
+    In [2]: ws.request_module_ownership('Empty')
+    In [3]: ws.administer({'command': 'listModRequests'})
+    Out[4]:
+    [{u'moduleName': u'Empty', ...}]
+    In [5]: ws.administer({'command': 'approveModRequest', 'module': 'Empty'})
+    In [6]: ws.register_typespec({'spec': spec, 'new_types': ['AType'], 'dryrun': 0})
+    Out[7]: {u'Empty.Atype-0.1': ....}
+    In [8]: ws.register_module('Empty')
+    Out[9]: [u'Empty.AType-1.0']
+    In [10]: ws.save_objects({'id': 1, 'objects': [{'type': 'Empty.AType', 'data': {'bar': 'baz'}, 'name': 'myobj'}]})
+    Out[11]:
+    [[1,
+    u'myobj',
+    ...
+    ]]
+
+
+Create a new terminal and start mongo to check to make sure the event has been written. Note that the status is UNPROC (unprocessed event).
+
+.. code-block:: bash
+
+    $ mongo
+    > show dbs
+    Search_test
+    admin
+    local
+    workspace
+    ws_types
+    > use Search_test
+    switched to db Search_test
+    > db.getCollectionNames()
+    ["searchEvents"]
+    > db.searchEvents.findOne()
+    {
+          "_id": ...,
+          "strcde": "WS",
+          "accgrp": 1,
+          ...
+          "status": "UNPROC"
+    }
+
+6. Create a new terminal and edit search_tools.cfg, create a test data type and build the executable script.
+
+.. code-block:: bash
+
+    $ cd [PATH_TO_YOUR_KBaseSearchEngine_DIR]
+    $ git checkout develop
+    $ git pull
+    $ cp search_tools.cfg.example search_tools.cfg
+    $ gedit search_tools.cfg
+
+Make the following edits.
+
+.. code-block:: cfg
+
+    search-mongo-host=localhost
+    search-mongo-db=Search_test
+    elastic-host=localhost
+    elastic-port=9200
+    scratch=/home/apasha/dev/kbase/KBaseSearchEngine/temp_scratch
+    workspace-url=http://localhost:7058
+    auth-service-url=https://ci.kbase.us/services/auth/api/legacy/KBase/Sessions/Login
+    indexer-token=[YOUR_CI_TOKEN]
```

Comment:
Note the user for this token must have workspace admin privs

Reply:
Added.